File: api/data/membership-schema/src/main/java/net/corda/schema/membership/provider/MembershipSchemaProviderFactory.java
package net.corda.schema.membership.provider;

import net.corda.schema.membership.provider.impl.MembershipSchemaProviderImpl;
import org.jetbrains.annotations.NotNull;

/**
 * Factory for membership schema providers.
 */
public final class MembershipSchemaProviderFactory {
    private MembershipSchemaProviderFactory() {
    }

    /**
     * Create a new membership schema provider.
     *
     * @return The new membership schema provider.
     */
    @NotNull
    public static MembershipSchemaProvider getSchemaProvider() {
        return new MembershipSchemaProviderImpl();
    }
}

File: data/membership-schema/src/main/kotlin/net/corda/schema/membership/provider/MembershipSchemaProviderFactory.kt
package net.corda.schema.membership.provider

import net.corda.schema.membership.provider.impl.MembershipSchemaProviderImpl

/**
 * Factory for membership schema providers.
 */
object MembershipSchemaProviderFactory {

    /**
     * Create a new membership schema provider.
     *
     * @return The new membership schema provider.
     */
    fun getSchemaProvider() : MembershipSchemaProvider {
        return MembershipSchemaProviderImpl()
    }
}